### PR TITLE
Add query "Global Server Object Uses HTTP" #2837

### LINKED ISSUE #2837: Add "Global Server Object Uses HTTP" query for OpenAPI
### Platform
*OpenAPI*

### Description
*Add "Global Server Object Uses HTTP" query for OpenAPI that checks if ULR's are not starting with HTTPS*

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
Closes #2837

**Proposed Changes**
- Add query "Global Server Object Uses HTTP" that checks if:
- servers global object is defined
- servers global object count > 0
- servers global object's URL uses HTTP

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

**Review by joaoReigota1 — COMMENTED** [review]
Is it a vulnerability that Global Server object isn't defined, since it isn't required?

File: assets/queries/openAPI/global_server_uses_http/query.rego [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package Cx
+
+import data.generic.openapi as openAPILib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openAPILib.checkOpenAPI(doc) != "undefined"
+    object.get(doc, "servers", "undefined") == "undefined"
```
Comment (felipe-avelar) [inline assets/queries/openAPI/global_server_uses_http/query.rego:8]: check indentation

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Check comments, please

**Comment by felipe-avelar** [comment]
> Is it a vulnerability that Global Server object isn't defined, since it isn't required?

I think it should be a vulnerability, since the default value for URL will be `/` as you can see on [open api object servers field definition](https://swagger.io/specification/#openapi-object)

**Comment by joaoReigota1** [comment]
@felipe-avelar @rogeriopeixotocx I believe we have a query to check if at least a server is defined meaning we will have 2 results for the same problem

File: assets/queries/openAPI/global_server_uses_http/query.rego (hunk 1 above)
Comment (rogeriopeixotocx) [reply assets/queries/openAPI/global_server_uses_http/query.rego:8]: @felipe-avelar  this is some kind of Github illusion, click on the "view file" button on the top right of the diff to see that indentation is correct

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**